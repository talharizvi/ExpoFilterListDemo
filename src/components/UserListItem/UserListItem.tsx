import React from "react";
import {Text, View, StyleSheet} from 'react-native';
import colors from "../../constants/colors";

interface UserListItemProps{
    name: String,
    userType: number,
    index?: number
}

const UserListItem: React.FC<UserListItemProps> = ({name, userType, index})=>{

 return(<View style={styles.container} testID={`user-${index}`}>
    <View style={styles.parentView}>
    
    <View style={styles.nameIconView}>
        <Text style={styles.nameIconText}>{name[0]}</Text>
    </View>

    <View>
        <Text style={styles.userNameText}>{name}</Text>
        <Text style={styles.userTypeText}>{userType === 0 ? 'Admin': 'Manager'}</Text>
    </View>
    </View>
    
    
 </View>)

}

const styles = StyleSheet.create({
    container:{
        marginBottom: 24,
        marginTop:8
    },
    parentView:{
        flexDirection: 'row'
    },
    nameIconView:{
        backgroundColor:colors.background, 
        marginRight:14, 
        borderRadius:8,
        justifyContent:'center',
        paddingHorizontal:12
    },
    nameIconText:{
        color:colors.iconText,
    },
    userNameText:{
        fontWeight: '600',
        fontSize: 14
    },
    userTypeText:{
        color:colors.userText,
    }
    
});

export default UserListItem;