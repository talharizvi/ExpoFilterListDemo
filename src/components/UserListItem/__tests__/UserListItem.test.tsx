import React from 'react';
import { render, screen } from '@testing-library/react-native';
import UserListItem from '../UserListItem';

describe('UserListItem component', () => {
  let name:string;

  beforeEach(() => {
    name = 'John Mill';
  });

  it('renders user name and user type for Admin', () => {
    render(<UserListItem name={name} userType={0} />);
    expect(screen.getByText(name)).toBeDefined();
    expect(screen.getByText('Admin')).toBeDefined();
  });

  it('renders user name and user type for Manager', () => {
    render(<UserListItem name={name} userType={1} />);
    expect(screen.getByText(name)).toBeDefined();
    expect(screen.getByText('Manager')).toBeDefined();
  });

});