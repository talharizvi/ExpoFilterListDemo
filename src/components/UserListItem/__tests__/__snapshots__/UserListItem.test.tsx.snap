// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserListItem component renders user name and user type for Admin 1`] = `
<View
  style={
    {
      "marginBottom": 24,
      "marginTop": 8,
    }
  }
  testID="user-undefined"
>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "rgba(0, 122, 255, 0.12)",
          "borderRadius": 8,
          "justifyContent": "center",
          "marginRight": 14,
          "paddingHorizontal": 12,
        }
      }
    >
      <Text
        style={
          {
            "color": "#5280eb",
          }
        }
      >
        J
      </Text>
    </View>
    <View>
      <Text
        style={
          {
            "fontSize": 14,
            "fontWeight": "600",
          }
        }
      >
        John Mill
      </Text>
      <Text
        style={
          {
            "color": "#5e5e5d",
          }
        }
      >
        Admin
      </Text>
    </View>
  </View>
</View>
`;

exports[`UserListItem component renders user name and user type for Manager 1`] = `
<View
  style={
    {
      "marginBottom": 24,
      "marginTop": 8,
    }
  }
  testID="user-undefined"
>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "rgba(0, 122, 255, 0.12)",
          "borderRadius": 8,
          "justifyContent": "center",
          "marginRight": 14,
          "paddingHorizontal": 12,
        }
      }
    >
      <Text
        style={
          {
            "color": "#5280eb",
          }
        }
      >
        J
      </Text>
    </View>
    <View>
      <Text
        style={
          {
            "fontSize": 14,
            "fontWeight": "600",
          }
        }
      >
        John Mill
      </Text>
      <Text
        style={
          {
            "color": "#5e5e5d",
          }
        }
      >
        Manager
      </Text>
    </View>
  </View>
</View>
`;
